pkg/deploy/controller/generictrigger/controller.go

```diff
@@ -36,183 +29,19 @@ type DeploymentTriggerController struct {
 	codec runtime.Codec
 }
 
-// fatalError is an error which can't be retried.
-type fatalError string
-
-func (e fatalError) Error() string {
-	return fmt.Sprintf("fatal error handling configuration: %s", string(e))
-}
-
 // Handle processes deployment triggers for a deployment config.
 func (c *DeploymentTriggerController) Handle(config *deployapi.DeploymentConfig) error {
 	if len(config.Spec.Triggers) == 0 || config.Spec.Paused {
```

Comment:
We are also requeueing on image stream updates: the reason we are using an imagestream reference index (it is the same reason that makes the image change controller redundant with this pull, that's why it's removed in https://github.com/openshift/origin/pull/9349/commits/a8eb662ba8b3051cb5e1562c1fe8c4abb19f4226).